stack-graphs/src/storage.rs

```diff
@@ -145,6 +123,102 @@ impl SQLiteWriter {
         file_exists(&self.conn, file, tag)
     }
 
+    /// Clean all data from the database.
+    pub fn clean_all(&mut self) -> Result<usize> {
+        self.conn.execute("BEGIN", [])?;
+        let count = self.clean_all_inner()?;
+        self.conn.execute("COMMIT", [])?;
```

Comment:
Can you use [`self.conn.transaction`](https://docs.rs/rusqlite/0.29.0/rusqlite/struct.Connection.html#method.transaction) here instead of executing the `BEGIN` and `COMMIT` directly?  In particular if `clean_all_inner` raises an error, I'm not convinced that the transaction will be rolled back properly as written.  That would require updating the other methods to take in a `&Connection` instead of using `self.conn` directly, so that they could operate on the transaction object that you get back.

Reply:
That's a nice pattern. I assumed transactions would be dropped if the connection was dropped, but I didn't consider the case where a caller might handle the error and try again or something. Your suggestion makes that more robust.

I propose to do that in a separate PR though. Several of the follow-up PRs change this code again, and porting such changes through the whole PR stack is a bit time-consuming.